### PR TITLE
ci: remove more bypass workflows

### PR DESCRIPTION
**Pull request by zmb3**
Remove bypass workflows for os-compatibility-test, unit-tests-helm, and unit-test-rust jobs.

### CODE REVIEW

**Review by espadolini — APPROVED** [review]
🎉

**Review by rosstimothy — APPROVED** [review]

File: .github/workflows/os-compatibility-test.yaml [hunk 1]
```diff
@@ -15,9 +15,31 @@ on:
       - '**/*.md*'
```
Comment (camscale) [inline .github/workflows/os-compatibility-test.yaml:15]: This missed removing the paths here.

Comment (camscale) [inline .github/workflows/os-compatibility-test.yaml:15]: Did you intend to remove `**/*.md*` from the ignore list?

Comment (zmb3) [reply .github/workflows/os-compatibility-test.yaml:15]: Yes, I did. Matching works a bit different with the paths-ignore job and the built-in `paths-ignore` behavior.

Comment (camscale) [reply .github/workflows/os-compatibility-test.yaml:15]: This still stands - not sure if you missed this comment with the other one I added to the same line - these `paths-ignore:` should be removed, right? We don't want github doing the filtering, as it is now done by dorny/paths-filter?

Comment (zmb3) [reply .github/workflows/os-compatibility-test.yaml:15]: Thanks! I did indeed miss this. You're right, these should be removed. I'll do that tomorrow.

**Review by r0mant — APPROVED** [review]

**Review by camscale — APPROVED** [review]

**Pull request merged**